src/model/schemas.py

```diff
@@ -27,20 +27,40 @@ class Meta:
         strict = True
 
 
+class MediumCompound(Schema):
+    id = fields.String(required=True)
+    # Note: namespace should match a namespace identifier from miriam.
```

Comment:
Could enforce this with a `OneOf` validator https://marshmallow.readthedocs.io/en/stable/api_reference.html#marshmallow.validate.OneOf

Reply:
We have this hard-coded in `memote.support.annotation` at the moment. Memote would be a pretty heavy dependency just for this, though. We could copy the list and try to remember to update it when the one in memote changes? (This list is already missing `'inchi'`.)

```python
METABOLITE_ANNOTATIONS = OrderedDict([
    ('pubchem.compound', re.compile(r"^\d+$")),
    ('kegg.compound', re.compile(r"^C\d+$")),
    ('seed.compound', re.compile(r"^cpd\d+$")),
    ('inchikey', re.compile(
        r"^[A-Z]{14}\-[A-Z]{10}(\-[A-Z])?")),
    ('chebi', re.compile(r"^CHEBI:\d+$")),
    ('hmdb', re.compile(r"^HMDB\d{5}$")),
    ('reactome', re.compile(
        r"(^R-[A-Z]{3}-[0-9]+(-[0-9]+)?$)|(^REACT_\d+(\.\d+)?$)")),
    ('metanetx.chemical', re.compile(r"^MNXM\d+$")),
    ('bigg.metabolite', re.compile(r"^[a-z_A-Z0-9]+$")),
    ('biocyc', re.compile(
        r"^[A-Z-0-9]+(?<!CHEBI)(\:)?[A-Za-z0-9+_.%-]+$"))
])
```

Reply:
Hmm, giving this some more thought, I'm not sure what there really is to gain from rejecting requests with an invalid namespace. Objects with an invalid namespace be handled the same as any an identifier with a valid but unrecognized namespace. (That is, for unrecognized medium compounds a warning will be issued, while unrecognized measurements will result in an error), so these cases will be discovered anyway. Since adding and maintaining such a list adds a bit of complexity, I'm leaning towards just accepting any string.

Reply:
Yes, please go ahead. This was more an idea of how to improve the service in future.